### PR TITLE
[Scan to Update Inventory M1] Flow when the stock is not managed

### PR DESCRIPTION
**Pull request by toupper**
Part of: #11415 

## Description

With this PR we implement flow when the stock is not managed for a product or variation in the quick inventory update screen. The error notice will follow when we merge the error notices PR, so it can be reused.

## Testing instructions

With a product whose stock is not managed.

1. Go to products
2. Tap on the barcode scanner button
3. Scan the product barcode.
4. The quick inventory screen is open. See that we hide the quantity button and the we show a button to manage the stock.
5. Tap on that button. A loading indicator replaces the button.
6. When the operation finishes, the quantity appears (0) and the user can update the stock quantity.

Try also with a product variation.

## Screenshots

https://github.com/woocommerce/woocommerce-ios/assets/1864060/54373a36-babd-4ab8-ad69-ad2f324b479f

---
- [X] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/Classes/ViewRelated/Products/UpdateInventory/UpdateProductInventoryViewModel.swift
Comment (iamgabrielma) [inline WooCommerce/Classes/ViewRelated/Products/UpdateInventory/UpdateProductInventoryViewModel.swift:136]: What do you think if we rename those to something like:
```
StockManagementState {
  case enabled
  case disabled
}
```

File: WooCommerce/Classes/ViewRelated/Products/UpdateInventory/UpdateProductInventoryView.swift
Comment (iamgabrielma) [inline WooCommerce/Classes/ViewRelated/Products/UpdateInventory/UpdateProductInventoryView.swift:92]: Should we wrap this into a do-catch? So we're sure the `try?` it does not fail silently.

File: WooCommerce/Classes/ViewRelated/Products/UpdateInventory/UpdateProductInventoryViewModel.swift
Comment (iamgabrielma) [inline WooCommerce/Classes/ViewRelated/Products/UpdateInventory/UpdateProductInventoryViewModel.swift]: Do we log the error here for the moment? Or we'll handle this differently?

File: WooCommerce/Classes/ViewRelated/Products/UpdateInventory/UpdateProductInventoryViewModel.swift
Comment (iamgabrielma) [inline WooCommerce/Classes/ViewRelated/Products/UpdateInventory/UpdateProductInventoryViewModel.swift]: Is this a method we'd like to move to `InventoryItem`? so we only need to call something like `inventoryItem.formattedStock()`

**Review by iamgabrielma — APPROVED** [review]
Works well! 🚢

File: WooCommerce/Classes/ViewRelated/Products/UpdateInventory/UpdateProductInventoryViewModel.swift
Comment (toupper) [reply WooCommerce/Classes/ViewRelated/Products/UpdateInventory/UpdateProductInventoryViewModel.swift:136]: Thanks for the suggestion @iamgabrielma! It's true that your suggestion makes it much simpler, but I wanted to hide logic details and only show view details. Therefore I called it `ViewMode` and made the name meaningful on what the view needs to know: `stockManagementNeedsToBeEnabled` -> we have to show the enable stock management button.

File: WooCommerce/Classes/ViewRelated/Products/UpdateInventory/UpdateProductInventoryView.swift
Comment (toupper) [reply WooCommerce/Classes/ViewRelated/Products/UpdateInventory/UpdateProductInventoryView.swift:92]: Yup, error PR is coming. As mentioned in the description, I wanted to wait until the error notice PR was merged so we can profit from that implementation.

File: WooCommerce/Classes/ViewRelated/Products/UpdateInventory/UpdateProductInventoryViewModel.swift
Comment (toupper) [reply WooCommerce/Classes/ViewRelated/Products/UpdateInventory/UpdateProductInventoryViewModel.swift]: The PR handling the error is coming, so it will be handled there.

File: WooCommerce/Classes/ViewRelated/Products/UpdateInventory/UpdateProductInventoryViewModel.swift
Comment (toupper) [reply WooCommerce/Classes/ViewRelated/Products/UpdateInventory/UpdateProductInventoryViewModel.swift]: Good suggestion @iamgabrielma! As this is just one line, and the logic is the same for all the protocol implementations, I'd prefer to leave it as it is., instead of having to create extra functions on the protocol and its implementations.

**Pull request merged**
